lib/rucio/core/monitor.py

```diff
@@ -238,58 +249,122 @@ def record_gauge(name, value, labels=None):
         gauge.set(value)
 
 
-def record_timer(name, time, labels=None):
+def record_timer(name: str,
+                 time: float,
+                 *,
+                 labels: Optional[Dict] = None,
+                 buckets: Iterable[float] = _HISTOGRAM_DEFAULT_BUCKETS
+                 ) -> None:
     """
-     Log timing information for a single stat (in miliseconds)
+    Log a time measurement.
 
     :param name: The name of the stat to be updated.
-    :param time: The time to log.
+    :param time: The time (in seconds) to log.
     :param labels: labels used to parametrize the metric
+    :param buckets: Optional iterable of histogram bucket separators.
     """
-    timing = TIMINGS.get(name)
-    if not timing:
-        TIMINGS[name] = timing = MultiTiming(statsd=name, labelnames=labels.keys() if labels else ())
+    if name not in HISTOGRAMS:
+        # TODO: shouldn't the dictionary key really be (name, labels.keys(), buckets)?
+        # do we need the dictionary at all, or could we re-construct this class each time its needed?
+        HISTOGRAMS[name] = MultiTiming(statsd=name, labelnames=labels.keys() if labels else (), buckets=buckets)
```

Comment:
I'm not quite sure why the `MultiTiming`s are cached in this dictionary, but to ensure correctness the cache key should probably be extended to include the other attributes as well, right?